pkg/daemon/ceph/osd/daemon_test.go

```diff
@@ -172,7 +173,95 @@ USEC_INITIALIZED=1128667
 	 `
 )
 
+func TestContainBluestoreSignature(t *testing.T) {
+	fileContainBluestore, err := os.Create("fileContainBluestore")
+	if err != nil {
+		t.Fatal("failed to create test file")
+	}
+	defer fileContainBluestore.Close()
+	defer os.Remove("fileContainBluestore")
+
+	_, err = fileContainBluestore.Write([]byte(bluestoreSignature))
+	if err != nil {
+		t.Fatal("failed to write test file")
+	}
+
+	fileNotContainBluestore, err := os.Create("fileNotContainBluestore")
+	if err != nil {
+		t.Fatal("failed to create test file")
+	}
+	defer fileNotContainBluestore.Close()
+	defer os.Remove("fileNotContainBluestore")
+
+	_, err = fileNotContainBluestore.Write([]byte("invalid signature"))
+	if err != nil {
+		t.Fatal("failed to write test file")
+	}
+
+	fileNotReadable, err := os.Create("fileNotReadable")
+	if err != nil {
+		t.Fatal("failed to create test file")
+	}
+	defer fileNotReadable.Close()
+	defer os.Remove("fileNotReadable")
+
+	err = os.Chmod("fileNotReadable", 0)
+	if err != nil {
+		t.Fatal("failed to set permission of test file")
+	}
+
+	cases := []struct {
+		device   *sys.LocalDisk
+		valid    bool
+		hasError bool
+	}{
+		{
+			device: &sys.LocalDisk{
+				Name:     "sda",
+				RealPath: "fileContainBluestore",
+			},
+			valid:    true,
+			hasError: false,
+		},
+		{
+			device: &sys.LocalDisk{
+				Name:     "sda",
+				RealPath: "fileNotContainBluestore",
+			},
+			valid:    false,
+			hasError: false,
+		},
+		{
+			device: &sys.LocalDisk{
+				Name:     "sda",
+				RealPath: "fileNotReadable",
+			},
+			valid:    false,
+			hasError: true,
+		},
+		{
+			device: &sys.LocalDisk{
+				Name:     "sda",
+				RealPath: "fileNotExist",
+			},
+			valid:    false,
+			hasError: true,
+		},
+	}
+
+	for _, c := range cases {
+		valid, err := containBluestoreSignature(c.device)
+
+		assert.Equal(t, c.valid, valid)
+		assert.Equal(t, c.hasError, err != nil, c.device.RealPath, err)
```

Comment:
How about this approach for readability (pseudocode)? As a single line it is just difficult to understand for me.
```
if c.hasError
  assert.Error(t, err)
else
  assert.NoError(t, err)
```